tools/c7n_org/c7n_org/cli.py

```diff
@@ -405,13 +405,13 @@ def _get_env_creds(session, region):
 
 
 def run_account_script(account, region, output_dir, debug, script_args):
-    try:
+    env = os.environ.copy()
+
+    if account.get('role') or account.get('profile'):
         session = get_session(account, "org-script", region)
-    except ClientError:
-        return 1
+        env.update(_get_env_creds(session, region))
 
-    env = os.environ.copy()
-    env.update(_get_env_creds(session, region))
+    env.update(account_tags(account))
```

Comment:
because this can cause tags to not be overlapped, they'll need to be restored to avoid any pollution between accounts. ie account A sets tags x, y, z.. Account B sets y, z.. account B can see env var for x.

Reply:
Env is created fresh for each account.. so pollution is impossible..
See https://github.com/cloud-custodian/cloud-custodian/pull/5130/files#diff-d0ad1d152e091597783f18cea409d819R408

Reply:
your right, thanks